addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitives.py

```diff
@@ -43,48 +44,69 @@ def gather_primitives(
+    primitives = []
+
     blender_primitives = gltf2_blender_extract.extract_primitives(
         None, blender_mesh, vertex_groups, modifiers, export_settings)
 
     for internal_primitive in blender_primitives:
-        if bpy.app.version < (2, 80, 0):
-            def __gather_materials_279(blender_primitive, blender_mesh, modifiers, export_settings):
-                if not blender_primitive['material']:
-                    # TODO: fix 'extract_primitives' so that the value of 'material' is None and not empty string
-                    return None
-                mesh_double_sided = blender_mesh.show_double_sided
-                material = bpy.data.materials[blender_primitive['material']]
-                return gltf2_blender_gather_materials.gather_material(material, mesh_double_sided, export_settings)
-
-            primitive = gltf2_io.MeshPrimitive(
-                attributes=__gather_attributes(internal_primitive, blender_mesh, modifiers, export_settings),
-                extensions=None,
-                extras=None,
-                indices=__gather_indices(internal_primitive, blender_mesh, modifiers, export_settings),
-                material=__gather_materials_279(internal_primitive, blender_mesh, modifiers, export_settings),
-                mode=None,
-                targets=__gather_targets(internal_primitive, blender_mesh, modifiers, export_settings)
-            )
-        else:
-            primitive = gltf2_io.MeshPrimitive(
-                attributes=__gather_attributes(internal_primitive, blender_mesh, modifiers, export_settings),
-                extensions=None,
-                extras=None,
-                indices=__gather_indices(internal_primitive, blender_mesh, modifiers, export_settings),
-                material=__gather_materials(internal_primitive, modifiers, export_settings),
-                mode=None,
-                targets=__gather_targets(internal_primitive, blender_mesh, modifiers, export_settings)
-            )
+        primitive = {
+            "attributes": __gather_attributes(internal_primitive, blender_mesh, modifiers, export_settings),
+            "indices": __gather_indices(internal_primitive, blender_mesh, modifiers, export_settings),
+            "material": internal_primitive['material'],
+            "targets": __gather_targets(internal_primitive, blender_mesh, modifiers, export_settings)
+        }
         primitives.append(primitive)
 
     return primitives
 
-
-def __gather_materials(blender_primitive, modifiers, export_settings):
-    if not blender_primitive['material']:
-        # TODO: fix 'extract_primitives' so that the value of 'material' is None and not empty string
+def __gather_materials(blender_primitive, material_names, modifiers, export_settings):
+    if blender_primitive['material'] is None:
         return None
-    material = bpy.data.materials[blender_primitive['material']]
+    # material = bpy.data.materials[blender_primitive['material']]
```

Comment:
Remove commented line.

Also, below, remove the `material_double_sided` parameter.  The `gather_material` function can work that out for itself now that a flag from the mesh no longer affects the material.

Reply:
Actually that whole function is not used anymore. Removed in latest commit